### DISCUSSION TITLE
Get YAxis tick values to calculate width

### DISCUSSION THREAD

**Discussion by dalvarez** [recharts/recharts#4445] [Q&A] [answered]
I'm trying to eliminate the gap that appears between the edge of the chart and the start of the YAxis labels, indicated by this red line i drew:
![CleanShot 2024-04-18 at 11 42 42@2x](https://github.com/recharts/recharts/assets/56429/33dbecac-fd41-4b52-bd87-ab20a4d80be4)

 I saw [this](https://github.com/recharts/recharts/issues/2027#issuecomment-769674096) as a possibility which makes sense, though still a bit of a hack to get it to work. However, since the library is calculating the actual labels for my number axis, I'm not sure how i can grab those values. 

Arbitrary example:
- my data set has values ranging from 25 to 87, and my YAxis is left to calculate the domain (default is `[0, 'auto']`) and this results in a chart of 0 to 100. How can i grab the range that the YAxis calculates for me so i can determine the hard coded width to apply to the YAxis?

My YAxis:
```        
        <YAxis
          axisLine={false}
          tickLine={false}
          tick={YTick}
          width={30}
        />
```
        
Custom Tick:

```
const YTick = (props: AxisProps) => {
  let tickValue = props.payload.value.toLocaleString();

  return (
    <Text
      {...props}
      x={props.width}
      textAnchor="end"
      verticalAnchor="end"
      fill="currentcolor"
      className="text-xs text-gray-600"
    >
      {tickValue}
    </Text>
  );
};
```

**Comment by ckifer** [comment, marked as answer]
Left offset is `margin` + `YAxis` width so make sure margin is accounted for (you have direct control over that).

You cannot (currently) easily grab the domain/range that recharts calculates for you. 

In 3.0 we will expose hooks that will allow you to get this information back - for now the only thing I could suggest leads to more hacks which you may not want to rely on.


You can use the `Customized` component to do all sorts of crazy things and access all/most of the internal state of recharts, but this will be going away in 3.0 with hooks in its place (as its hard to document and maintain). There you have access to what the entire left offset value is. The codesandbox should show using it, but in your case you'll have to save the value in state somehow and then come back and update your axis width

Example here
[![Edit custom-reference-areas](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-line-chart-forked-j5jyjw?file=%2Fsrc%2FApp.tsx%3A79%2C11)

**Reply by dalvarez** [reply]
Ah gotcha, thanks for the explanation, example and insight to whats coming next!

For the margin on the `<YAxis />` is that an exported prop? I didnt see one in the API docs so wasn't sure.

**Reply by ckifer** [reply]
No there's no margin on `YAxis` itself - I just meant on the chart item. i.e. `<LineChart margin={{ left: 0 }} >...` - by default there is some units of margin there

**Reply by dalvarez** [reply]
understood, thanks!

**Comment by briavicenti** [comment]
Hey @dalvarez, I'm hoping to accomplish something similar in my own project. Were you able to use the `Customized` component to get the width you need? I haven't been able to get this setup working yet and would really appreciate any tips you have.

**Reply by ckifer** [reply]
@briavicenti can you add a sandbox with what you have? Anything in particular you are struggling with?

**Reply by briavicenti** [reply]
@ckifer I appreciate you hopping in here so quickly but I spoke too soon and happened to have a breakthrough here a few minutes ago. The TLDR of my issue was that I was using the `Customized` component to render a hidden `text` element for measuring the necessary label width and was struggling to observe changes to the `text` element due to limitations in the observer APIs for SVG elements (SO [here](https://stackoverflow.com/questions/65565149/how-to-apply-resizeobserver-to-svg-element)).

I then swapped to using [this approach](https://stackoverflow.com/a/21015393) for measuring text and so am all good now. My setup is pretty funky though, I've essentially created a `Customized` component whose sole purpose is just using other hooks to gather/set state and renders nothing. I'm eager to use the hooks you describe above instead!

**Reply by ckifer** [reply]
got it! fingers crossed we can get it out soon - slowing down a bit during the holidays but hoping to pick it back up soon enough
